cmake/ChooseBLAS.cmake

```diff
@@ -0,0 +1,273 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set(BLAS "Open" CACHE STRING "Selected BLAS library")
+set_property(CACHE BLAS PROPERTY STRINGS "Atlas;Open;MKL;Apple")
+
+function(switch_lapack ENABLE)
+  if(ENABLE)
+    message(STATUS "Enabling LAPACK functionality")
+    add_definitions(-DMXNET_USE_LAPACK=1)
+  else()
+    if(USE_LAPACK)
+      message(WARNING "LAPACK functionality not available")
+    else()
+      message(STATUS "LAPACK functionality not available")
+    endif()
+  endif()
+endfunction()
+
+function(try_mkldnn)
+
+  if(NOT USE_MKLDNN)
+    return()
+  endif()
+
+  if(NOT MKL_FOUND)
+    message(WARNING "MKLDNN requires either MKL or MKLML, MKLDNN will not be available")
```

Comment:
My understanding was that MKLDNN could be installed stand-alone, without any other MKL flavours.

Reply:
I was actually the opposite opinion, but after taking a closer look I've realised that this is true. I will make the needed changes.